src/main/java/org/testeditor/fixture/web/json/BrowserSettingsManager.java

```diff
@@ -42,81 +43,78 @@
         return browserSetupElementsOsSpecific;
     }
-    
-    protected Map<String,Object> copyOptionsIntoMap(Map<String,Object> browserSettingsMap,
-        List<BrowserSetupElement> browserSetupElements) {
-        browserSetupElements.forEach((bsetting) -> { 
-                bsetting.getOptions().forEach((option) ->  {
-                    String key = option.getKey();
-                    if (browserSettingsMap.containsKey(key)) {
-                        throw new RuntimeException("Duplicate entries existing in configuration file " 
-                            + FILE_NAME + ". Entries are: [" + key + " - " + option.getValue() + ", " 
-                            + key + " - " + browserSettingsMap.get(key) + "]");
-                    } else {
-                        browserSettingsMap.put(key, option.getValue());
-                    }
-                } 
-                );
-        }   
-        ); 
+
+    protected Map<String, Object> copyOptionsIntoMap(Map<String, Object> browserSettingsMap,
+            List<BrowserSetupElement> browserSetupElements) throws FixtureException {
+        for (BrowserSetupElement bsetting : browserSetupElements) {
+            for (BrowserSetting option : bsetting.getOptions()) {
+                String key = option.getKey();
+                if (browserSettingsMap.containsKey(key)) {
+                    throw new FixtureException("Duplicate entries existing in configuration file",
+                            WebDriverFixture.keyValues("file", FILE_NAME, //
+                                    "offendingKey", key, //
+                                    "valueOption", option.getValue(), //
+                                    "valueBrowserSettingsMap", browserSettingsMap.get(key)));
+                } else {
+                    browserSettingsMap.put(key, option.getValue());
+                }
+            }
+        }
         return browserSettingsMap;
     }
-    
-    protected Map<String,Object> copyCapabilitiesIntoMap(Map<String,Object> browserSettingsMap,
-        List<BrowserSetupElement> separateBrowserSettingsOsspecific) {
-        separateBrowserSettingsOsspecific.forEach((bsetting) -> { 
-                bsetting.getCapabilities().forEach((capability) ->  { 
-                    String key = capability.getKey();
-                    if (browserSettingsMap.containsKey(key)) {
-                        throw new RuntimeException("Duplicate entries existing in configuration file " 
-                            + FILE_NAME + ". Entries are: [" + key + " - " + capability.getValue() + ", " 
-                            + key + " - " + browserSettingsMap.get(key) + "]");
-                    } else {
-                        browserSettingsMap.put(key, capability.getValue());
-                    }
-                } 
-                );
-        }   
-        ); 
+
+    protected Map<String, Object> copyCapabilitiesIntoMap(Map<String, Object> browserSettingsMap,
```

Comment:
not sure if worth the effort, but given the similarities with `copyOptionsIntoMap` – extract a generic method?

Reply:
didn't want to touch that, actually, but you are right. I'll take a look at it